### ISSUE TITLE
Is there a plan to release the WindLayer?

### ISSUE THREAD

**Issue by Robert-OP** [visgl/deck.gl#4348]
Hi there, I am really impressed about the [wind demo](https://github.com/uber/deck.gl/tree/6.4-release/showcases/wind).

Q1: Are there any plans on releasing the [WindLayer](https://github.com/uber/deck.gl/blob/6.4-release/showcases/wind/src/layers/wind-layer/wind-layer.js)?

Q2: Was also looking [how the data is fetched](https://github.com/uber/deck.gl/blob/6.4-release/showcases/wind/src/utils/load-data.js). I see that it is weather.bin, but I'm curious - how does the data looks like?

Cheers

**Comment by ibgreen** [comment]
> Are there any plans on releasing the WindLayer?

It has certainly been a long-standing ambition, but so far we have felt that we'd need a more comprehensive treatment of data textures, e.g. to be able to provide a clear answer to your second question.

In the mean-time, if you are willing to a bit of extra work, you can always copy the layer from the example into your app.

>  Was also looking how the data is fetched. I see that it is weather.bin, but I'm curious - how does the data looks like?

I don't recall exactly at the moment, but I believe that if you look into the shader code, you can rather easily see that e.g wind speed in each point is encoded in color components of the resulting data texture.

**Comment by Robert-OP** [comment]
Cheers @ibgreen - thanks a lot for your time!

Great work that you guys are doing, hope I can help out!

**Comment by Robert-OP** [comment]
@ibgreen I was actually looking to implement the WindLayer, but with the newest deck.gl and luma.gl libraries, because now it depends on older versions. 

Would you see any problem/obstacles that I would encounter by using the newest versions?

Cheers

**Comment by ibgreen** [comment]
@Robert-OP 

> I was actually looking to implement the WindLayer, but with the newest deck.gl and luma.gl libraries, because now it depends on older versions.

Yes, that would be a great first step towards making these layers generally usable.

Q1: Would you see any problem/obstacles that I would encounter by using the newest versions?

Yes, there would likely be some effort involved, as APIs have changed a bit on the lower levels.

Q2: Is it possible to implement something similar, e.g. WindLayer ParticleLayer, with the newer version?

Yes of course, it would just be a matter of working around some API changes in deck and luma.

Before you start, you'd definitely want to pay close attention to deck.gl and luma.gl upgrade guides.

Some of it you could probably do through taking the existing example on a branch and just upgrading the deck.gl versions, and we could help you work through any console errors and warnings.

I believe there were some shader changes in 5.0->6.0 (maybe around picking uniforms) that resulted in silent failures, those could be hard for you to debug on your own, but you could get advice here.

Perhaps @1chandu may have better recollections of what was changed.

**Comment by Robert-OP** [comment]
Thanks @ibgreen! 

I would love to help out, work and contribute on it, but I would need some guidance.

I have already started using the things from the wind demo showcase and put them in my project, but I had to modify and go through some errors and warnings (I bumped up the version of deck.gl@6.1.0 and luma.gl@6.4.2 - I want to go higher now and make them work with the latest versions).

![image](https://user-images.githubusercontent.com/16807022/76874725-e58b6d80-686f-11ea-8194-00284195112d.png)
